namer/namer.cc

```diff
@@ -2001,14 +2000,20 @@ class TreeSymbolizer {
         // Prevent a crash by validating the arguments before we do any symbol lookups or manipulations
         if (send->hasPosArgs() || send->hasKwArgs() || send->block() != nullptr) {
             if (send->numPosArgs() > 1) {
-                if (auto e = ctx.beginError(send->loc, core::errors::Namer::InvalidTypeDefinition)) {
-                    e.setHeader("Too many args in type definition");
+                auto extraArgsLoc = send->getPosArg(1).loc().join(send->getPosArg(send->numPosArgs() - 1).loc());
+                if (auto e = ctx.beginError(extraArgsLoc, core::errors::Namer::InvalidTypeDefinition)) {
+                    e.setHeader("Too many arguments in `{}` definition for `{}`", send->fun.show(ctx),
+                                typeName->cnst.show(ctx));
+                    auto firstArgLoc = send->getPosArg(0).loc();
+                    auto lastArgLoc = send->getPosArg(send->numPosArgs() - 1).loc();
+                    auto replacementRange = core::Loc(ctx.file, firstArgLoc.endPos(), lastArgLoc.endPos());
+                    e.replaceWith("Retain only the first argument", replacementRange, "{}({})", send->fun.show(ctx),
+                                  send->getPosArg(0).toString(ctx));
```

Comment:
`toString` is a method we use for printing an AST in debugging output. It is not guaranteed to produce valid source code (for the case where the tree node represents a symbol literal, it happens to work).

Better is to just repeat whatever the user wrote for that arg

```suggestion
                                  ctx.locAt(firstArgLoc).source(ctx);
```